service/src/test/java/com/epam/rft/atsy/service/impl/PositionServiceImplTest.java

```diff
@@ -1,69 +1,186 @@
 package com.epam.rft.atsy.service.impl;
 
-/**
- * Unit tests for {@link PositionServiceImpl}.
- */
-/*public class PositionServiceImplTest {
-    @InjectMocks
-    private PositionServiceImpl underTest;
-    @Mock
-    private PositionDAO dao;
+import com.epam.rft.atsy.persistence.entities.PositionEntity;
+import com.epam.rft.atsy.persistence.repositories.PositionRepository;
+import com.epam.rft.atsy.service.domain.PositionDTO;
+import com.epam.rft.atsy.service.exception.DuplicateRecordException;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Matchers;
+import org.hibernate.exception.ConstraintViolationException;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.TypeToken;
+import org.springframework.dao.DataIntegrityViolationException;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.times;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PositionServiceImplTest {
     @Mock
     private ModelMapper modelMapper;
 
-    @BeforeMethod
-    public void init() {
-        MockitoAnnotations.initMocks(this);
+    @Mock
+    private PositionRepository positionRepository;
+
+    @InjectMocks
+    private PositionServiceImpl positionService;
+
+    private static Type POSITIONDTO_LIST_TYPE;
+
+    private static final Long DEVELOPER_ID = 1L;
+    private static final String DEVELOPER_NAME = "Developer";
+
+    private static PositionEntity DEVELOPER_ENTITY;
+    private static PositionDTO DEVELOPER_DTO;
+
+    private static final List<PositionEntity> EMPTY_POSITION_ENTITY_LIST = Collections.emptyList();
+    private static final Collection<PositionDTO> EMPTY_POSITION_DTO_LIST = Collections.emptyList();
+
+    private static List<PositionEntity> EXPECTED_POSITION_ENTITY_LIST;
+    private static Collection<PositionDTO> EXPECTED_POSITION_DTO_LIST;
+
+    @BeforeClass
+    public static void setUp() {
+        POSITIONDTO_LIST_TYPE = new TypeToken<List<PositionDTO>>() {}.getType();
+
```

Comment:
The fields in the setup block should not be uppercase, as they are not strictly constants. Only POSITIONDTO_LIST_TYPE  should remain uppercase, and it should be defined outside of the setUp method. 

It's hard to draw the line between what should be uppercase, but there is a good quote from google's recommendation: "Every constant is a static final field, but not all static final fields are constants. Before choosing constant case, consider whether the field really feels like a constant. For example, if any of that instance's observable state can change, it is almost certainly not a constant. Merely intending to never mutate the object is generally not enough."